`block-spacing` errors on multi line blocks

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.2
* **npm Version:** 4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<!-- Paste your configuration below: -->
.eslintrc.js
```js
module.exports = {
    'parserOptions': {
        'ecmaVersion': 2017
    },
    'env': {
        'node': true,
        'es6': true
    },
    'root': true,
    'extends': 'eslint:recommended',
    'rules': {
        'block-spacing': 'error',
		
        //turn off other rules to make output clean
        'max-statements-per-line': 'off',
        'no-unused-vars': 'off'
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
The docs describe the `block-spacing` rule as "Disallow or enforce spaces inside of single line blocks". However, the rule is not limited to single line blocks. It is applied multi line blocks as well.

For example, both functions  below trigger the `block-spacing` rule.

<!-- Paste the source code below: -->
```js
'use strict';
function test() {let i = 0;
    return i;
}

function test2() {//comment
    return 0;
}
```

**What did you expect to happen?**
no errors from `block-spacing` rule

**What actually happened? Please include the actual, raw output from ESLint.**
  2:17  error  Requires a space after '{'  block-spacing
  6:18  error  Requires a space after '{'  block-spacing

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.

--- a/docs/rules/block-spacing.md
+++ b/docs/rules/block-spacing.md
@@ -1,4 +1,4 @@
-# Disallow or enforce spaces on the same line between an open block token and the next token, and between a closing block token and the previous token (block-spacing)
+# Disallow or enforce spaces inside of blocks after opening block and before closing block (block-spacing)
 
 ## Rule Details
 

--- a/docs/rules/block-spacing.md
+++ b/docs/rules/block-spacing.md
@@ -1,8 +1,8 @@
-# Disallow or enforce spaces between an open block token and the next token on the same line (block-spacing)
+# Disallow or enforce spaces on the same line between an open block token and the next token, and between a closing block token and the previous token (block-spacing)
 
 ## Rule Details
 
-This rule enforces consistent spacing inside an open block token and the next token on the same line. Also, this rule enforces consistent spacing inside a close block token and previous token on the same line.
+This rule enforces consistent spacing inside an open block token and the next token on the same line. This rule also enforces consistent spacing inside a close block token and previous token on the same line.
 
 ## Options
 

--- a/docs/rules/block-spacing.md
+++ b/docs/rules/block-spacing.md
@@ -1,8 +1,8 @@
-# Disallow or enforce spaces inside of single line blocks (block-spacing)
+# Disallow or enforce spaces between an open block token and the next token on the same line (block-spacing)
 
 ## Rule Details
 
-This rule enforces consistent spacing inside single-line blocks.
+This rule enforces consistent spacing inside an open block token and the next token on the same line. Also, this rule enforces consistent spacing inside a close block token and previous token on the same line.
 
 ## Options
 
@@ -20,6 +20,9 @@ Examples of **incorrect** code for this rule with the default `"always"` option:
 
 function foo() {return true;}
 if (foo) { bar = 0;}
+function baz() {let i = 0;
+    return i;
+}
 ```
 
 Examples of **correct** code for this rule with the default `"always"` option:

--- a/lib/rules/block-spacing.js
+++ b/lib/rules/block-spacing.js
@@ -14,7 +14,7 @@ const util = require("../ast-utils");
 module.exports = {
     meta: {
         docs: {
-            description: "enforce consistent spacing inside single-line blocks",
+            description: "enforce consistent spacing on the same line between an open block token and the next token, and between a closing block token and the previous token",
             category: "Stylistic Issues",
             recommended: false
         },

--- a/lib/rules/block-spacing.js
+++ b/lib/rules/block-spacing.js
@@ -14,7 +14,7 @@ const util = require("../ast-utils");
 module.exports = {
     meta: {
         docs: {
-            description: "Disallow or enforce spaces inside of blocks after opening block and before closing block",
+            description: "disallow or enforce spaces inside of blocks after opening block and before closing block",
             category: "Stylistic Issues",
             recommended: false
         },

--- a/lib/rules/block-spacing.js
+++ b/lib/rules/block-spacing.js
@@ -14,7 +14,7 @@ const util = require("../ast-utils");
 module.exports = {
     meta: {
         docs: {
-            description: "enforce consistent spacing on the same line between an open block token and the next token, and between a closing block token and the previous token",
+            description: "Disallow or enforce spaces inside of blocks after opening block and before closing block",
             category: "Stylistic Issues",
             recommended: false
         },

